internal/pipe/checksums/checksums.go

```diff
@@ -40,14 +40,19 @@ func (Pipe) Run(ctx *context.Context) (err error) {
 	if ctx.Config.Checksum.Disable {
 		return pipe.Skip("checksum.disable is set")
 	}
-	artifactList := ctx.Artifacts.Filter(
-		artifact.Or(
+	var filter artifact.Filter
+	if len(ctx.Config.Checksum.IDs) == 0 {
+		filter = artifact.Or(
 			artifact.ByType(artifact.UploadableArchive),
 			artifact.ByType(artifact.UploadableBinary),
 			artifact.ByType(artifact.UploadableSourceArchive),
 			artifact.ByType(artifact.LinuxPackage),
-		),
-	).List()
+		)
+	} else {
+		filter = artifact.ByIDs(ctx.Config.Checksum.IDs...)
+	}
```

Comment:
I would say this should be an `and` (ids && filters by type)

Reply:
You're right. I was under the assumption that the proposed behaviour was in line with how other ID filters work, but looking at it again, that does not seem to be the case. I changed it.